tests/spec/jasmine-event-recur.js

```diff
@@ -226,6 +233,27 @@ describe("The Calendar Interval", function() {
         expect(valentines.matches( moment(startDate) )).toBe(false);
     });
 
+    it("weeksOfMonthByDay should work combined with daysOfWeek", function() {
+        var recurrence = moment.recur().every(["Sunday"]).daysOfWeek().every([0, 2]).weeksOfMonthByDay();
+        expect(recurrence.matches(moment(startDate))).toBe(false);
+        expect(recurrence.matches(moment(startDate).date(6))).toBe(true);
+        expect(recurrence.matches(moment(startDate).date(8))).toBe(false);
+        expect(recurrence.matches(moment(startDate).date(13))).toBe(false);
+        expect(recurrence.matches(moment(startDate).date(20))).toBe(true);
+        expect(recurrence.matches(moment(startDate).date(27))).toBe(false);
+
+        recurrence = moment.recur().every(["Sunday", "Thursday"]).daysOfWeek().every([1, 3, 4]).weeksOfMonthByDay();
```

Comment:
Sounds good. It seemed that the `weeksOfMonth` rule was using a zero-based index for the week of the month. Because of that convention, `weeksOfMonthByDay` is zero-indexed.

```
// week === 0
var week = moment('2013/01/01', 'YYYY/MM/DD').monthWeek();
```

1/31/2015 is the 5th Thursday of the month.

I'll probably go with `it("recurs on the 2nd, 4th and 5th Sunday & Thursday of the month")`.